homeassistant/components/pvpc_hourly_pricing/sensor.py

```diff
@@ -137,14 +167,26 @@ def __init__(
         super().__init__(coordinator)
         self.entity_description = description
         self._attr_attribution = coordinator.api.attribution
-        self._attr_unique_id = unique_id
+        self._attr_unique_id = make_sensor_unique_id(unique_id, description.key)
```

Comment:
This should not be in an external API

Reply:
Hi @emontnemery, I see. The logic in the external lib is this: 
(there are 2 methods to convert between sensor key and HA unique-ids)

```python
_ha_uniqueid_to_sensor_key = {
    TARIFFS[0]: KEY_PVPC,
    TARIFFS[1]: KEY_PVPC,
    f"{TARIFFS[0]}_{KEY_INJECTION}": KEY_INJECTION,
    f"{TARIFFS[1]}_{KEY_INJECTION}": KEY_INJECTION,
    f"{TARIFFS[0]}_{KEY_MAG}": KEY_MAG,
    f"{TARIFFS[1]}_{KEY_MAG}": KEY_MAG,
    f"{TARIFFS[0]}_{KEY_OMIE}": KEY_OMIE,
    f"{TARIFFS[1]}_{KEY_OMIE}": KEY_OMIE,
}


def get_enabled_sensor_keys(
    using_private_api: bool, disabled_sensor_ids: list[str]
) -> set[str]:
    """(HA) Get enabled API indicators."""
    sensor_keys = set(ALL_SENSORS) if using_private_api else {KEY_PVPC}
    for unique_id in disabled_sensor_ids:
        disabled_ind = _ha_uniqueid_to_sensor_key[unique_id]
        if disabled_ind in sensor_keys:
            sensor_keys.remove(disabled_ind)

    return sensor_keys


def make_sensor_unique_id(config_entry_id: str, sensor_key: str) -> str:
    """(HA) Generate unique_id for each sensor kind and config entry."""
    assert sensor_key in ALL_SENSORS
    if sensor_key == KEY_PVPC:
        # for old compatibility
        return config_entry_id
    return f"{config_entry_id}_{sensor_key}"
```

As you can see, the old `unique_id` was just the selected tariff, just to be able to configure an entry for each tariff. To handle the addition of new sensors I'm just composing tariff + sensor-key, but leaving the old one as it was, for backward compatibility. 

Should I resolve this logic in other way? Or it's ok if I just move those methods to the integration?

Reply:
Move this logic to the integration please 👍